File: src/chapter3/code3.3.ts
function code3_3(){
    //線形探索法
    //aの中から最小値を求める
    const a = [4,3,12,7,2,11]
    let INF = 2000000 //十分に大きな値
    let min_value = INF

    for (let i = 0; i < a.length; i++) {
        if(a[i] < min_value){
            min_value = a[i] //最小値を更新
        }
    }
    
    console.log(min_value)
}
code3_3()

File: src/chapter3/code3.3.cpp
#include <iostream>
#include <vector>
using namespace std;
//最小値を求める
const int INF = 2000000; //十分大きい値で初期化

int main(){
    //入力を受け取る
    int N, v;
    cin >> N >> v;
    vector<int> a(N);
    for(int i = 0; i < N; ++i) cin >> a[i];

    //線形探索
    int min_value = INF;
    for (int i = 0; i < N; i++)
    {
        if(a[i] < min_value) {
            min_value = a[i];
        }
    }
    
    //結果出力
    cout << min_value << endl;
}